backend/functions/lib/adapters/gateways/document-db.repository.ts

```diff
@@ -0,0 +1,56 @@
+export type FilterOperation = {
+  equals?: unknown;
+  greaterThan?: unknown;
+  greaterThanOrEqual?: unknown;
+  contains?: unknown;
+  lessThan?: unknown;
+  lessThanOrEqual?: unknown;
+  notEqual?: unknown;
+  notContains?: unknown;
+  not?: unknown;
+  exists?: boolean;
+};
+
+export type Filter = FilterOperation & {
+  [key: string]: unknown;
+};
+
+export type BooleanOperation = {
+  and?: Filter[];
+  or?: Filter[];
+  all?: unknown[];
+};
+
+export type DocumentQuery = BooleanOperation & {
```

Comment:
IF all other code in this file is removed then this type should be moved or refactored into a better file.